frontend/packages/pipelines-plugin/src/components/pipelines/pipeline-overview/pipeline-overview-utils.ts

```diff
@@ -0,0 +1,22 @@
+import { PIPELINE_RUN_AUTO_START_FAILED } from '@console/dev-console/src/components/topology/redux/const';
+
+export const getNotStartedPipelines = () => {
+  return JSON.parse(sessionStorage.getItem(PIPELINE_RUN_AUTO_START_FAILED) ?? '[]');
+};
+export const isPipelineNotStarted = (pipelineName: string): boolean => {
+  return getNotStartedPipelines().includes(pipelineName);
+};
+
+export const removePipelineNotStarted = (pipelineName: string): void => {
+  const newList = getNotStartedPipelines().filter((pName) => pName !== pipelineName);
+  sessionStorage.setItem(PIPELINE_RUN_AUTO_START_FAILED, JSON.stringify(newList));
+};
+
+export const setPipelineNotStarted = (pipelineName: string): void => {
+  if (!pipelineName) return;
+  const pipelineList = getNotStartedPipelines();
+  if (pipelineList.length === 0 || !pipelineList.includes(pipelineName)) {
+    pipelineList.push(pipelineName);
+  }
+  sessionStorage.setItem(PIPELINE_RUN_AUTO_START_FAILED, JSON.stringify(pipelineList));
+};
```

Comment:
We want to hold onto the namespace as well... if you just do name, I could switch between projects and get unexpected results.

Plus you probably want to only write it back out when you modify it... no need to write the same value you got back to the storage if it didn't change.

```suggestion
export const setPipelineNotStarted = (namespace: string, pipelineName: string): void => {
  if (!pipelineName || !namespace) return;

  const pipelineData = getNotStartedPipelines();
  if (!pipelineData[namespace]) {
    pipelineData[namespace] = [];
  }
  if (!pipelineData[namespace].includes(pipelineName)) {
    pipelineData[namespace].push(pipelineName);
    sessionStorage.setItem(PIPELINE_RUN_AUTO_START_FAILED, JSON.stringify(pipelineData));
  }
};
```